### PR TITLE
ensure strict json parsing in JSON functions

### PR DESCRIPTION
**Pull request by walterddr**
## Description

ensure string json parsing, so that errors such as duplicate key issues will not occur.

## Additional context and related issues

address #22329

## Release notes

(x) Release notes are required. Please propose a release note for me.

### CODE REVIEW

**Comment by mosabua** [comment]
@martint ... quick glance and potentially reopen?

**Comment by kasiafi** [comment]
Not reopening this PR.  cc @mosabua  
Here's why:

Currently the `json_parse` function removes duplicate keys in JSON objects.
Note that this function is not part of ISO SQL spec. However, the spec defines the json-parsing operation (6.40 \<JSON parse\>, p. 357), and I will use it for reference.

The default behavior for keys is `WITHOUT UNIQUE KEYS`:
```
If <JSON key uniqueness constraint> is not specified, then WITHOUT UNIQUE KEYS is implicit.
```
It is further explained (9.42 Parsing JSON text, p. 722):
```
It is implementation-dependent (UA057) whether members with redundant duplicate keys are removed. If the implementation-dependent (UA057) choice is to delete members with redundant duplicate keys, then for all i, 1 (one) ≤ i ≤ m, and all j, 1 (one) ≤ j ≤ m, i ≠ j, if Ki is equivalent to Kj, an implementation-dependent (UA057) choice of Mi or Mj is removed from the list of members of J.
```
TL;DR: the current duplicate key handling in the `json_parse` function is legit, while failing on duplicates isn't.

Also, changing this would be backwards-incompatible.

Hopefully, we will cover all JSON operations as described in the SQL spec in the future. It will let the user control the key uniqueness behavior.

**Comment by mosabua** [comment]
Thank you for the clarification @kasiafi !

**Comment by walterddr** [comment]
Thank you for the clarification @kasiafi 

Could you suggest an alternative to detect duplicate key in json structure?

According to https://trino.io/docs/current/functions/json.html#json-path-syntax-and-semantics it states 
I don't see any function we can detect duplicates keys.

I understand this is backward-incompatible change but I was new so only proposing the simplest solution. 

Can we consider adding optional `WITH UNIQUE KEY` argument to json_parse, or a json_validate function?

**Comment by kasiafi** [comment]
@walterddr currently, only the `json_object` function supports detecting duplicate keys. With this function, you can detect duplicates when building a JSON object, but not when parsing. The spec proposes json parse function and `IS [NOT] JSON` predicate, both with optional duplicate key detection. I hope we will be able to ad those soon, but it's not quite straightforward, since we first need to make sure our JSON type implementation is compliant with the spec.

**Comment by walterddr** [comment]
yeah `json_object` is probably not the right function i can use since it is cannot decode a stringify json but generate a json object from the input. but yeah looking forward to whatever solution fit until then i will just use a patched docker image with this PR which i've been doing for a while.
